kdumpctl

```diff
@@ -1659,47 +1662,156 @@ _should_reset_crashkernel() {
+		exit 2
+		;;
+	*)
+		# Found one or more files containing the test id
+		dir=\$(dirname \$id_file | uniq)
+		case \$(echo -n "\$dir" | wc -w) in
+			1)
+				# The test id files within one folder
+				;;
+			*)
+				# The test id files shouldn't within
+				# multiple folders
+				exit 2
+				;;
+		esac
+		;;
+esac
+if [[ -f "\$dir/vmcore" || -f "\$dir/vmcore.flat" ]]; then
+	# Success
+	exit 0
+else
+	# Fail
+	exit 1
+fi
+EOF
+	if is_nfs_dump_target || is_local_target; then
+		mkdir -p "$TMPMNT"
+		mount "${OPT[_target]}" "$TMPMNT" -t "${OPT[_fstype]}" -o defaults || \
+			{ dwarn "Failed to mount ${OPT[_target]}" && return 2; }
+		bash "$_script" "$TMPMNT/${OPT[path]}"
+	elif is_ssh_dump_target; then
+		ssh -i "${OPT[sshkey]}" -o BatchMode=yes "${OPT[_target]}" \
+			'bash -s' -- < "$_script" "${OPT[path]}"
+	elif is_raw_dump_target; then
+		return 2
+	else
+		bash "$_script" "${OPT[path]}"
+	fi
+}
+
 check_vmcore_creation_status()
 {
-	local _status _timestamp _status_date
+	local _status _test_id _timestamp _status_date
 
 	[[ ${VMCORE_CREATION_NOTIFICATION,,} == "yes" ]] || return
 
-	if [[ ! -s $VMCORE_CREATION_STATUS ]]; then
+	if [[ ! -s "$VMCORE_CREATION_STATUS" ]]; then
 		dwarn "Notice: No vmcore creation test performed!"
 		return
 	fi
 
-	read -r _status _timestamp < "$VMCORE_CREATION_STATUS"
-	_status_date="$(date -d "@$_timestamp")"
+	[[ "${#OPT[@]}" -eq 0 ]] && { parse_config || return; }
 
-	if [[ "$_status" == "success" ]]; then
+	read -r _status _test_id < "$VMCORE_CREATION_STATUS"
+	_timestamp="$(echo "$_test_id" | awk -F '[-=]' '{print $2}')"
+	_status_date=$(date -d "@$_timestamp")
+
+	if [[ "$_status" == "pending" ]]; then
+		fetch_test_status "$_test_id"
+		case "$?" in
+		0)
+			set_vmcore_creation_status "success"
+			dinfo "Notice: Last successful vmcore creation on $_status_date"
+			;;
+		1)
+			set_vmcore_creation_status "fail"
+			dwarn "Notice: Last NOT successful vmcore creation on $_status_date"
+			;;
+		*)
+			set_vmcore_creation_status "manual"
+			dwarn "Notice: Require manual check for kdump test of $_status_date"
+			;;
+		esac
+	elif [[ "$_status" == "success" ]]; then
 		dinfo "Notice: Last successful vmcore creation on $_status_date"
-	else
+	elif [[ "$_status" == "fail" ]]; then
 		dwarn "Notice: Last NOT successful vmcore creation on $_status_date"
+	elif [[ "$_status" == "manual" ]]; then
+		dwarn "Notice: Require manual check for kdump test of $_status_date"
 	fi
 }
 
 kdump_test()
 {
-	local _dir
+	local _kdump_test_id="kdump_test_id=$(date +%s-%N)"
 
 	if ! is_kernel_loaded "$DEFAULT_DUMP_MODE"; then
 		derror "Kdump needs be operational before test."
 		exit 1
 	fi
 
-	_dir=$(dirname "$VMCORE_CREATION_STATUS")
-	if ! [[ -d "$_dir" ]]; then
-		derror "Vmcore status dir $_dir not exist."
-		exit 1
-	fi
-
-	if ! lsblk $(get_mount_info SOURCE target "$_dir") > /dev/null; then
-		derror "$VMCORE_CREATION_STATUS must on local drive"
-		exit 1
-	fi
-
 	if [[ ! "$1" == "--force" ]]; then
```

Comment:
@liutgnu There is a $1, but seems no arguments are passed.

Reply:
Correct, the argument is missing, "$@" need to passed.